### PR TITLE
Setup GitHub Actions for CI

### PR DESCRIPTION
**Pull request by subhransu**
Setup GitHub Actions to Trigger build for PR and when pushed to master. 

At this moment this PR sets up:

* Necessary packages
* Makes a docker build 
* Tags the docker build
* And pushes to a private docker repository 

## Further work is still remaining to host te docker build once this PR is merged.

### CODE REVIEW

**Comment by subhransu** [comment]
Build fails as the secrets are not passed to PRs created from a fork.
Successful build can be found here: https://github.com/subhransu/jobsgowhere/runs/571512423

**Comment by kajalsinha** [comment]
https://stackoverflow.com/questions/30970591/automatic-docker-login-within-a-bash-script/30971862 <-- this might help to fix docker login issue @subhransu

**Comment by subhransu** [comment]
@kajalsinha I am using the same [here](https://github.com/jobsgowhere/jobsgowhere/pull/39/files#diff-90829e76e906f1c73140c7ded7e1b268R27)

but the secrets won't be passed if it is a PR from a forked repo. 

This workflow is working properly so we just need to merge the PR and then it will work once it's merged.

File: Makefile [hunk 1]
```diff
@@ -26,3 +25,6 @@ docker-build: build-ui
 
 docker-run:
 	docker run -p 8080:8080 jobsgowhere/server
+
+docker-pus:
```
Comment (ivanfoong) [inline Makefile:29]: `docker-pus` typo?

File: Makefile [hunk 2]
```diff
@@ -13,7 +11,8 @@ clean:
 build-full: clean build
 
 build-ui:
-	cd ui && npm run build
+	npm install webpack-dev-server rimraf webpack react-scripts -g
```
Comment (ivanfoong) [inline Makefile:14]: is this Github Action specific? or its required by the react build process? If required by build process, should we just add it into frontend `package.json`

**Review by ivanfoong — CHANGES_REQUESTED** [review]

**Comment by subhransu** [comment]
Closing this PR as created a branch on this repo and sent #48 
This addresses the changes requested on this PR as well.

**Pull request closed without merging**